scripts/common.lib.ps1

```diff
@@ -51,15 +53,25 @@ $env:DOTNET_CLI_VERSION = $GlobalJson.tools.dotnet
 $env:VSWHERE_VERSION = "2.0.2"
 $env:MSBUILD_VERSION = "15.0"
 
-function Write-Log ([string] $message)
-{
+function Write-Log {
+    param (
+        [string] $message,
+        [ValidateSet("Success", "Error")]
+        [string]
+        $Level = "Success"
+    )
+    
     $currentColor = $Host.UI.RawUI.ForegroundColor
-    $Host.UI.RawUI.ForegroundColor = "Green"
-    if ($message)
-    {
-        Write-Output "... $message"
+    try {
+        $Host.UI.RawUI.ForegroundColor = if ("Success" -eq $Level) { "Green" } else { "Red" }
+        if ($message)
```

Comment:
Could we maybe do some kind of early return here and basically do nothing, not even change the `ForegroundColor` if the `$message` variable is empty ? Something along the lines of:

```
function Write-Log {
    param (...)

    if (!$message)
    {
        return
    }

    try {
        ...
    }
    finally {
        ...
    }
```

Reply:
We might, but it won't save any time. Thanks for the feedback, but I won't do it now.